boa/src/syntax/parser/expression/assignment/mod.rs

```diff
@@ -105,25 +113,29 @@ impl TokenParser for AssignmentExpression {
                     return Ok(node);
                 }
             }
+
             _ => {}
         }
 
+        cursor.set_goal(InputElement::Div);
+
         let mut lhs = ConditionalExpression::new(self.allow_in, self.allow_yield, self.allow_await)
             .parse(cursor)?;
 
-        if let Some(tok) = cursor.next() {
-            match tok.kind {
+        if let Some(tok) = cursor.peek() {
+            match tok?.kind() {
```

Comment:
These statements (I believe?) have a different meaning, .peek() returns an Option and we don't want to ? the option we want to ? the token inside the option.

Reply:
I see. then we can use `.transpose()` this will convert `Option<io::Result<u8>>` into `io::Result<Option<u8>>` and then we can use the try operator.
```rust
	if let Some(tok) = cursor.peek().transpose()? {
            match tok.kind() {
```

Or should we restructure the way we return from `.next()` and `.peek()` instead of  
`Option<io::Result<u8>>`, we return a `Option<u8>` this would simplefy the code, do we really need to have the io error or can we ignore it?

Reply:
We need the IO error because a call to parser::Cursor.peek() may invoke the lexer which may trigger an IO error.

Re: transpose I'll make the change